src/publish.js

```diff
@@ -12,7 +12,7 @@ module.exports = async function publish(pluginConfig, {
     const mavenTarget = pluginConfig.mavenTarget || 'deploy';
     const clean = pluginConfig.clean || true;
 
-    if (!/^[\w~./]*$/.test(settingsPath)) {
+    if (!/^[\w~./-]*$/.test(settingsPath)) {
```

Comment:
This regex fix that allows dashes in path must be applied to all other places where this check is done.
By the way, I created a PR to remove this check that doesn't provide any value from my point of view.

Reply:
See below. Regarding the fix in general vs. removal, perhaps @simonseyock has an opinion here?

Reply:
From my point of view, this should be removed as security checks are made by maven itself. No need for this plugin to enrich tests with this regexp that will be hard to tune to cover all cases. For instance, the current regexp doesn't allow `\`, which I agree is quite weird in a path name, but that could exist though.

Reply:
I introduced the checks because I was not totally sure that it would be a good idea to allow unsanitized input that will be passed into `exec`. The value comes form a config file, though.

I am not entirely sure, consider this scenario:
A repository is using this plugin on PR. A user inserts malicious code into the config file that prints out secrets. The pipeline will run with the changed config and the secrets are leaked.

I don't know if I am overly paranoid. Is the scenario realistic? How is this handled in other github actions?

In the end it would be nice if we could get rid of these checks because it is annoying to maintain them.

Reply:
@simonseyock I understand this can be a concern, but it should be the responsability of the exec function to sanitize its arguments, not yours. Did you check by the way that `execa` that is used underneath doesn't already do some verifications ?
At least the fixed regexp is still too restrictive I think: it doesn't support `\` in it. There will always be someone who will use a specific value for settings.xml path that won't work with the current impl.